scripts/generate-icon-exports.js

```diff
@@ -9,101 +9,161 @@ const { stripIndents } = require('common-tags');
+shelljs.exec(
+  `rm -rf ${publicRawExportsPath} ${publicIconComponentsPath} ${internalRawExportsPath} ${internalIconComponentsPath}`
+);
 
-      Component.displayName = '${displayName}';
+shelljs.exec(
+  `mkdir ${publicRawExportsPath} ${publicIconComponentsPath} ${internalRawExportsPath} ${internalIconComponentsPath}`
+);
 
-      // we do this to enable treeshaking
-      // please see https://github.com/alex996/react-css-spinners/issues/1
-      if (process.env.NODE_ENV !== "production") {
-        Component.propTypes = iconPropTypes;
-      }
+const iconsToProcess = [
+  {
+    indexPath: publicIndexPath,
+    svgPath: publicIconsPath,
+    rawExportsPath: publicRawExportsPath,
+    iconComponentsPath: publicIconComponentsPath,
+  },
+  {
+    indexPath: internalIndexPath,
+    svgPath: internalIconsPath,
+    rawExportsPath: internalRawExportsPath,
+    iconComponentsPath: internalIconComponentsPath,
+  },
+];
+
+iconsToProcess.forEach(data => {
+  glob(data.svgPath, async (err, files) => {
+    const relativeStyledIconPath = path.relative(
+      data.iconComponentsPath,
+      createStyledIconPath
+    );
 
-      export default Component;
+    files.forEach(fileName => {
+      const fileNameWithoutExtension = path.basename(fileName, iconFileExt);
+      const componentName = upperFirst(camelCase(fileNameWithoutExtension));
+
+      const svgCode = fs.readFileSync(fileName, 'UTF-8');
+      const jsCode = svgr.sync(
+        svgCode,
+        {
+          icon: false,
+          svgoConfig: {
+            plugins: [
+              { removeViewBox: false },
+              { prefixIds: true },
+              // same result as rollup plugin
+              { prefixIds: { prefix: fileNameWithoutExtension } },
+
+              // Keeps ID's of svgs so they can be targeted with CSS
+              { cleanupIDs: false },
+            ],
+          },
+          // same as the rollup plugin
+          plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx'],
+        },
+        { componentName }
+      );
+
+      fs.writeFileSync(
+        path.join(data.rawExportsPath, `/${fileNameWithoutExtension}.js`),
+        prettier.format(jsCode, prettierConfig)
+      );
+
+      const displayName = `${componentName}Icon`;
+
+      const importStatement = stripIndents`
+        import React from 'react';
+        import { getIconStyles, iconPropTypes } from '${relativeStyledIconPath}';
+        import ${componentName} from '../raw-components/${fileNameWithoutExtension}';
+
+        const Component = props => <${componentName} {...props} css={theme => getIconStyles(props, theme)} />;
+
+        Component.displayName = '${displayName}';
+
+        // we do this to enable treeshaking
+        // please see https://github.com/alex996/react-css-spinners/issues/1
+        if (process.env.NODE_ENV !== "production") {
+          Component.propTypes = iconPropTypes;
+        }
+
+        export default Component;
+      `;
```

Comment:
Starting to wonder if we could some time dump them into a `icon.template.js` or seomthing.

Reply:
people: redux is too much boilerplate! everything is in different files.
also people: hey, we have 3 things in one file. let's put it in three.

😄

Reply:
I see your point but was never the one who hated Redux for boilerplate :)